ui/src/logs/components/LogsTable.tsx

```diff
@@ -120,22 +131,75 @@ class LogsTable extends Component<Props, State> {
     )
   }
 
+  private handleWindowResize = () => {
+    this.currentMessageWidth = null
+    this.grid.current.recomputeGridSize()
+    this.headerGrid.current.recomputeGridSize()
+  }
+
   private handleHeaderScroll = ({scrollLeft}) => this.setState({scrollLeft})
 
   private handleScrollbarScroll = (e: MouseEvent<JSX.Element>) => {
     const {target} = e
     this.handleScroll(target)
   }
 
+  private get widthMapping() {
+    return {
+      timestamp: 160,
+      procid: 80,
+      facility: 120,
+      severity: 22,
+      severity_1: 120,
+      host: 300,
+    }
+  }
+
+  private get messageWidth() {
+    if (this.currentMessageWidth) {
+      return this.currentMessageWidth
+    }
+
+    const columns = getDeep<string[]>(this.props, 'data.columns', [])
+    const otherWidth = columns.reduce((acc, col) => {
+      if (col === 'message' || col === 'time') {
+        return acc
+      }
+
+      return acc + _.get(this.widthMapping, col, 200)
+    }, 0)
+
+    const calculatedWidth = window.innerWidth - (otherWidth + 180)
+    this.currentMessageWidth = Math.max(100 * CHAR_WIDTH, calculatedWidth)
+
+    return this.currentMessageWidth - CHAR_WIDTH
+  }
+
+  private getColumnWidth = ({index}: {index: number}) => {
+    const column = getDeep<string>(this.props, `data.columns.${index + 1}`, '')
+
+    switch (column) {
+      case 'message':
+        return this.messageWidth
+      default:
+        return _.get(this.widthMapping, column, 200)
```

Comment:
I see `_.get(this.widthMapping, col, 200)` used here and above. It might be a good getter method